Tavisca.Bootcamp.LanguageBasics.Exercise2/Program.cs

```diff
@@ -13,18 +13,80 @@ static void Main(string[] args)
-        {
+        private static void Test(string[] postTimes, string[] showTimes, string expected)
+        {
             var result = GetCurrentTime(postTimes, showTimes).Equals(expected) ? "PASS" : "FAIL";
             var postTimesCsv = string.Join(", ", postTimes);
             var showTimesCsv = string.Join(", ", showTimes);
-            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
+            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
         }
 
-        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
+        {
+            // Author : Aditi Rupade.
+
+            TimeSpan currentTime = new TimeSpan(0),  // to store current times for each iteration
+                 posttime  = new TimeSpan(0),     // to store parsed post times
+                 latervalue = new TimeSpan(0);    // to store time after adding the message times
+
+            // for same post times but showing different messages
+            if (exactPostTimes.Distinct().Count() != showPostTimes.Distinct().Count())
+                return "impossible";
+
+            // iterating for each value in exactPostTimes array
+            for (int i = 0; i < exactPostTimes.Length; i++)
+            {
+                //parsing and splitting 
+                posttime = TimeSpan.Parse(exactPostTimes[i]);
+                string[] val = showPostTimes[i].Split(' ');
+
+
+                //Case 1: Checking for seconds
+                if (val[1] =="seconds")
+                {
+                    if ( posttime > currentTime)
+                        currentTime = posttime;
+
+                }
+                else
+                {
+                    int value = int.Parse(val[0]);
+
+                    //Case 2: Checking for minutes
+                    if (val[1].Contains("min"))
+                    {
+                        latervalue = TimeSpan.FromMinutes(value);
+
+                        currentTime = TimeSpan.Parse(GetCurrentValue(latervalue, posttime, currentTime));
+                    }
+
+                    //Case 3: checking for hours
+                    if (val[i].Contains("hour"))
+                    {
+                        latervalue = TimeSpan.FromHours(value);
+
+                        currentTime = TimeSpan.Parse(GetCurrentValue(latervalue, posttime, currentTime));
+                    }
+                }
+            }
+            return currentTime.ToString();
+        }
+        
+        public static string GetCurrentValue(TimeSpan latervalue, TimeSpan posttime, TimeSpan currentTime)
+        {
+            //if current time is next day
+            if (posttime + latervalue > TimeSpan.Parse("1.00:00:00"))

```

Comment:
`posttime + latervalue` can be used as a variable with meaningful name to avoid duplicate usages